presto-main/src/main/java/com/facebook/presto/execution/AddColumnTask.java

```diff
@@ -62,7 +63,8 @@ public String getName()
         Session session = stateMachine.getSession();
         QualifiedObjectName tableName = createQualifiedObjectName(session, statement, statement.getName());
         Optional<TableHandle> tableHandle = metadata.getTableHandle(session, tableName);
-        if (!tableHandle.isPresent()) {
+        Optional<ConnectorMaterializedViewDefinition> optionalMaterializedView = metadata.getMaterializedView(session, tableName);
+        if (!tableHandle.isPresent() || optionalMaterializedView.isPresent()) {
```

Comment:
This doesn't look right. We shouldn't throw MISSING_TABLE error. We should tell user we cannot alter a materialized view.